src/coreclr/jit/hwintrinsic.h

```diff
@@ -614,8 +614,41 @@ struct HWIntrinsicInfo
         switch (id)
         {
             case NI_AVX512F_Add:
+            case NI_AVX512F_Divide:
+            case NI_AVX512F_Multiply:
+            case NI_AVX512F_Subtract:
+
+            case NI_AVX512F_AddScalar:
+            case NI_AVX512F_DivideScalar:
+            case NI_AVX512F_MultiplyScalar:
+            case NI_AVX512F_SubtractScalar:
+            case NI_AVX512F_SqrtScalar:
```

Comment:
Is this one correct? `SqrtScalar` typically has two overloads exposed:
```csharp
public static Vector128<float> SqrtScalar(Vector128<float> value);
public static Vector128<float> SqrtScalar(Vector128<float> upper, Vector128<float> value);
```

So I'd expect the rounding support to also be either `2` or `3` parameters. Additionally since the `Avx512F` versions of the `scalar` APIs always take an immediate as the last operand, I would expect simply getting the operand count to be sufficient for any of these `NI_AVX512F_*Scalar` entries.

Reply:
In the approved APIs surfaces (#73604), the `SqrtScalar` related APIs all take 2 operands plus the embedded rounding control byte. I am not sure if I missed this version somewhere: `public static Vector128<float> SqrtScalar(Vector128<float> value);`.

And yes, I agree the point that it would be enough to just get the operand count to detect embedded rounding for these intrinsics, I will try to figure out how I can simplify the checks.

Reply:
That looks like a missed consistency thing. We've always exposed both variants where the 1 parameter overload uses `value` implicitly as `upper`. I'd expect we want and need to support it here as well

Reply:
Sorry I might be missing something, for `SSE/SSE2`, yes we do have 2 overloads for `sqrtscalar`. But for `Avx512F`, I was not able to find the specific definition for this version, `SqrtScalar(value)`.

Reply:
I was saying I think we missed it in API review. However, we're going to end up wanting to add that for parity long term (I'll work on getting a proposal up to add it), so we should ensure this logic is setup to support that.

Reply:
Ah, I see